pkg/agent/plugin/nodeattestor/aws/iid.go

```diff
@@ -109,6 +112,48 @@ func fetchMetadata(endpoint string) (*caws.IIDAttestationData, error) {
 	}, nil
 }
 
+func getMetadataDoc(ctx context.Context, client *imds.Client) (string, error) {
+	res, err := client.GetDynamicData(ctx, &imds.GetDynamicDataInput{
+		Path: docPath,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	doc, err := readStringAndClose(res.Content)
+	if err != nil {
+		return "", err
+	}
+
+	return doc, nil
+}
+
+func getMetadataSig(ctx context.Context, client *imds.Client) (string, error) {
+	res, err := client.GetDynamicData(ctx, &imds.GetDynamicDataInput{
+		Path: sigPath,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	sig, err := readStringAndClose(res.Content)
+	if err != nil {
+		return "", err
+	}
+
+	return sig, nil
+}
+
+func readStringAndClose(r io.ReadCloser) (res string, err error) {
+	defer r.Close()
+	var sb strings.Builder
+	if _, err := io.Copy(&sb, r); err == nil {
+		res = sb.String()
+	}
+
+	return
+}
```

Comment:
Sorry, I had a comment here. I think it got swallowed when I tried to submit during last weeks outage. We should avoid naked returns:
```suggestion
func readStringAndClose(r io.ReadCloser) (string, error) {
	defer r.Close()
	var sb strings.Builder
	if _, err := io.Copy(&sb, r); err == nil {
		return sb.String(), nil
	} else {
		return "", err
	}
}
```
or alternatively:
```suggestion
func readStringAndClose(r io.ReadCloser) (string, error) {
	defer r.Close()
	var sb strings.Builder
	_, err := io.Copy(&sb, r)
	return sb.String(), err
}
```